### ISSUE TITLE
Source branch name validation missing

### ISSUE THREAD

**Issue by szymonrychu** [mergeability/mergeable#424]
I really really need to check source branch name in a pull request.
Something like `baseRef` but for the branch names that the merge request is based on to ensure compatibility with other systems/integrations like Jira.

Example of config:
```
version: 2
mergeable:
  - when: pull_request.*
    validate:
      - do: ref
        must_include:
          regex: '^(feature|hotfix)\/.+$'
          message: |
            Your pull request doesn't adhere to the branch naming convention described <a href="some link">there</a>!'
```
Based on provided example branch named `feature/something` or `hotfix/XD-123` will be passing.
Branch named simply `xd` will fail with verbose description on why and how to improve the MR.

**Comment by szymonrychu** [comment]
Right now I need to use custom github action to have this functionality:
```
name: branch-name-check

on:
  pull_request:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest
    steps:
    - uses: mshick/add-pr-comment@v1
      with:
        message: |
          Your pull request doesn't adhere to the convention described <a href="some link">there</a>!
        repo-token: ${{ secrets.GITHUB_TOKEN }}
        repo-token-user-login: 'github-actions[bot]' # The user.login for temporary GitHub tokens
        allow-repeats: true # This is the default
    - name: 'Fail on wrong branch name!'
      if: github.ref != 'refs/heads/feature/.*' && github.ref != 'refs/heads/hotfix/.*'
      run: |
        exit 1
```

**Comment by jusx** [comment]
@szymonrychu Yup. I can see how this can be useful.

**Comment by shine2lay** [comment]
@szymonrychu hi, we recently merged a baseRef validator that i think would fit in your use case. Here is the the example https://github.com/mergeability/mergeable/pull/421/files#diff-b15a1ccd9c27ec69a0e22cafc6895ee42e2352b326c6769641de2af557d58d7dR1 (sorry, it doesn't display properly in current docs because of missing connection, we'll fix it in a bit)

**Comment by roydenwa** [comment]
@ all I would call the validator headRef, because I think that the payload of a pull_request also contains an object called `head` with the needed `ref`.

**Comment by jusx** [comment]
I was thinking `branch` but `headRef` is a much more consistent naming convention and aligns with `baseRef` nicely.

**Comment by szymonrychu** [comment]
These all things might be a bit problematic, becasue of how Github Actions are handling contents of `github.ref`. 
The variable depends on when the checks are running. When GHA runs "on pull_request" it consists of `refs/pull/NO/head` or something similar, which is not matching the original `refs/heads/<name of the branch>`- it's somewhat special to Github- it simulates what would happen after a merge to base. :)

**Comment by shine2lay** [comment]
implemented at #439 . Regarding the GH Action, i think we should tackle it when we get to it.
